src/daft-dsl/src/functions/utf8/left.rs

```diff
@@ -0,0 +1,50 @@
+use crate::Expr;
+use daft_core::{
+    datatypes::{DataType, Field},
+    schema::Schema,
+    series::Series,
+};
+
+use common_error::{DaftError, DaftResult};
+
+use super::super::FunctionEvaluator;
+
+pub(super) struct LeftEvaluator {}
+
+impl FunctionEvaluator for LeftEvaluator {
+    fn fn_name(&self) -> &'static str {
+        "left"
+    }
+
+    fn to_field(&self, inputs: &[Expr], schema: &Schema, _: &Expr) -> DaftResult<Field> {
+        match inputs {
+            [data, pattern] => match (data.to_field(schema), pattern.to_field(schema)) {
+                (Ok(data_field), Ok(pattern_field)) => {
+                    match (&data_field.dtype, &pattern_field.dtype) {
+                        (DataType::Utf8, DataType::UInt32) => {
```

Comment:
We should also support other integer datatypes, such as UInt64, Int32, Int64, etc.